This patch is based on the upstream commit described below, adapted for use
in the Debian package by Peter Michael Green.

commit 67d1aae529fc1bec9d8d728dc6f875381aa32054
Author: Guillaume Gomez <guillaume1.gomez@gmail.com>
Date:   Tue Aug 9 00:45:41 2022 +0200

    Update sysinfo version

Index: process-viewer/src/display_sysinfo.rs
===================================================================
--- process-viewer.orig/src/display_sysinfo.rs
+++ process-viewer/src/display_sysinfo.rs
@@ -3,7 +3,7 @@ use gtk::prelude::{
     AdjustmentExt, BoxExt, ContainerExt, GridExt, LabelExt, ProgressBarExt, ScrolledWindowExt,
     ToggleButtonExt, WidgetExt,
 };
-use sysinfo::{self, ComponentExt, ProcessorExt, SystemExt};
+use sysinfo::{self, ComponentExt, CpuExt, SystemExt};
 
 use std::cell::RefCell;
 use std::iter;
@@ -173,13 +173,13 @@ impl DisplaySysInfo {
             p.set_margin_end(5);
             p.set_margin_start(5);
             p.set_show_text(true);
-            let processor = sys.global_processor_info();
+            let processor = sys.global_cpu_info();
             p.set_text(Some(&format!("{:.1} %", processor.cpu_usage())));
             p.set_fraction(f64::from(processor.cpu_usage() / 100.));
             vertical_layout.add(p);
         }
         let check_box = create_header("Processors usage", &vertical_layout, settings.display_graph);
-        for (i, pro) in sys.processors().iter().enumerate() {
+        for (i, pro) in sys.cpus().iter().enumerate() {
             procs.push(gtk::ProgressBar::new());
             let p: &gtk::ProgressBar = &procs[i + 1];
             let l = gtk::Label::new(Some(&format!("{}", i)));
@@ -414,13 +414,10 @@ impl DisplaySysInfo {
         let v = &*self.procs.borrow_mut();
         let h = &mut *self.cpu_usage_history.borrow_mut();
 
-        v[0].set_text(Some(&format!(
-            "{:.1} %",
-            sys.global_processor_info().cpu_usage()
-        )));
+        v[0].set_text(Some(&format!("{:.1} %", sys.global_cpu_info().cpu_usage())));
         v[0].set_show_text(true);
-        v[0].set_fraction(f64::from(sys.global_processor_info().cpu_usage() / 100.));
-        for (i, pro) in sys.processors().iter().enumerate() {
+        v[0].set_fraction(f64::from(sys.global_cpu_info().cpu_usage() / 100.));
+        for (i, pro) in sys.cpus().iter().enumerate() {
             let i = i + 1;
             v[i].set_text(Some(&format!("{:.1} %", pro.cpu_usage())));
             v[i].set_show_text(true);
Index: process-viewer/Cargo.toml
===================================================================
--- process-viewer.orig/Cargo.toml
+++ process-viewer/Cargo.toml
@@ -33,7 +33,7 @@ version = "1.0"
 version = "1.0"
 
 [dependencies.sysinfo]
-version = "0.23.0"
+version = "0.25.1"
 
 [dependencies.toml]
 version = "0.5"
